### PR TITLE
[npu] add npu support for hybrid plugin and llama

### PR DESCRIPTION
**Pull request by oahzxl**
## 📌 Checklist before creating the PR

- [ ] I have created an issue for this PR for traceability
- [x] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [x] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

## 📝 What does this PR do?

> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.

add npu support for hybrid plugin and llama

## 💥 Checklist before requesting a review

- [ ] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [x] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [x] I have performed a self-review of my code
- [ ] I have added thorough tests.
- [x] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [ ] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

**Comment by oahzxl** [comment]
image
cd ColossalAI/examples/language/llama2
torchrun --nproc_per_node 8 --standalone benchmark.py -g -b 2 -p 3d --zero 1  --tp 2

**Comment by oahzxl** [comment]
support: zero0,1,2, tp
not support: pp

File: colossalai/utils/device.py [hunk 1]
```diff
@@ -44,6 +44,18 @@ def get_current_device() -> torch.device:
         return torch.device("cpu")
 
 
+def get_current_device_id() -> int:
```
Comment (ver217) [inline colossalai/utils/device.py:47]: It seems this is similar to `current_device()` in this file

Comment (oahzxl) [reply colossalai/utils/device.py:47]: Yes, it returens "int" and current_device returns "device". Device index is required sometimes.

Comment (ver217) [reply colossalai/utils/device.py:47]: `current_device()` returns int, and `get_current_device()` returns device

Comment (oahzxl) [reply colossalai/utils/device.py]: ok, i have removed it

**Review by ver217 — APPROVED** [review]

**Pull request merged**